ambry-router/src/main/java/com/github/ambry/router/RouterCallback.java

```diff
@@ -60,5 +62,17 @@ void scheduleDeletes(List<StoreKey> idsToDelete, String serviceIdSuffix) {
       backgroundDeleteRequests.add(new BackgroundDeleteRequest(storeKey, serviceIdSuffix));
     }
   }
+
+  /**
+   * Schedule the deletes of ids in the given list.
+   * @param idsToDelete the set of blob ids that need to be deleted.
+   * @param serviceIdSuffix the suffix to append to the service ID when deleting these blobs.
+   */
+  void scheduleDeletesForBlobIds(Set<BlobId> idsToDelete, String serviceIdSuffix) {
```

Comment:
maybe we can rename this to scheduleSlippedDeletes()? In fact we might not even need this method if we didn't change the `BackgroundDeleteRequest` class (see comments above).